packages/test/test-utils/src/loaderContainerTracker.ts

```diff
@@ -167,8 +167,10 @@ export class LoaderContainerTracker implements IOpProcessingController {
             if (containersToApply.length === 0) { break; }
 
             // Ignore readonly dirty containers, because it can't sent up and nothing can be done about it being dirty
-            const dirtyContainers =
-                containersToApply.filter((c) => c.deltaManager.readOnlyInfo.readonly !== true && c.isDirty);
+            const dirtyContainers = containersToApply.filter((c) => {
+                const { deltaManager, attachState,isDirty } = c;
+                return deltaManager.readOnlyInfo.readonly !== true && attachState === AttachState.Attached && isDirty;
```

Comment:
I'm not sure why this change is necessary and why / if it is correct.
I'd think that usage of this API is limited to only attached containers, i.e., tests should ensure attachment process is finished before calling this API. If not, there is only one container (with unfinished file creation) to sync and calling this API and it returning right away does not give caller anything useful.

Reply:
I'll retest this after dirty flow correction. If isDirty is decoupled from Attach logic, this will probably go away.

Reply:
This change has been reverted.